### PR TITLE
Add Tests for Jobs and Client

### PR DESCRIPTION
**Pull request by ccfromms**
## Description
While testing importOrchestratorJob, I discovered several issues in job tests. Essentially the issues are as following:

- Need more tests for OrchestratorJob such as testing different job status combinations.
- The function GetJobByGroupId in TestQueueClient only read jobInfo from local variable.
- Missing test when job was cancelled by user and job was cancelled by itself at the same time, it is better to verify the job status in this situation.

This PR will add several tests for OrchestratorJob and SqlQueueClient, and upgrade the function for TestQueueClient.

## Testing

- Added some tests for OrchestratorJob:
  - Verify the thrown exception when some jobs cancelled or failed.
  - Verify the thrown exception and job status when job was cancelled or failed after several calls.
  - Verify the job status when job was cancelled or failed and other jobs were running or created.
 - Upgraded TestQueueClient, added func for GetJobByGroupId.
 - Added test for SqlQueueClientTest to verify job status after cancelJobByGroupId called twice.

## FHIR Team Checklist
- [X] **Update the title** of the PR to be succinct and less than 50 characters
- [X] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [X] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [ ] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [ ] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs [hunk 1]
```diff
@@ -259,6 +260,40 @@ public async Task GivenGroupJobs_WhenCancelJobsByGroupId_ThenAllJobsShouldBeCanc
             Assert.Equal(jobInfo2.Result, jobInfo.Result);
         }
 
+        [Fact]
+        public async Task GivenGroupJobs_WhenCancelJobsByGroupIdCalledTwoTimes_ThenJobStatusShouldNotChange()
```
Comment (tongwu-msft) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:264]: >TwoTimes [](http://example.com/codeflow?start=70&length=8)

Twice

Comment (ccfromms) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:264]: Sorry, I will modify the name of the function.

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs [hunk 2]
```diff
@@ -259,6 +260,40 @@ public async Task GivenGroupJobs_WhenCancelJobsByGroupId_ThenAllJobsShouldBeCanc
...
+            jobInfo2 = await sqlQueueClient.GetJobByIdAsync(queueType, 2, false, CancellationToken.None);
+            JobInfo jobInfo3 = await sqlQueueClient.GetJobByIdAsync(queueType, 3, false, CancellationToken.None);
+            Assert.Equal(JobStatus.Failed, jobInfo1.Status);
+            Assert.Equal(JobStatus.Running, jobInfo2.Status);
+            Assert.Equal(JobStatus.Cancelled, jobInfo3.Status);
+
+            await sqlQueueClient.CancelJobByGroupIdAsync(queueType, jobInfo2.GroupId, CancellationToken.None);
+            Assert.True((await sqlQueueClient.GetJobByGroupIdAsync(queueType, jobInfo2.GroupId, false, CancellationToken.None)).All(t => t.Status == JobStatus.Cancelled || t.Status == JobStatus.Failed || (t.Status == JobStatus.Running && t.CancelRequested)));
+            jobInfo1 = await sqlQueueClient.GetJobByIdAsync(queueType, 1, false, CancellationToken.None);
+            jobInfo2 = await sqlQueueClient.GetJobByIdAsync(queueType, 2, false, CancellationToken.None);
```
Comment (tongwu-msft) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:290]: >2 [](http://example.com/codeflow?start=71&length=1)

dequeue operation do not guarantee

Comment (ccfromms) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:290]: Sorry, I will check and replace number with variable.

**Comment by fhibf** [comment]
Hi! Please, sync this PR with main. 
I've added new validations on top of tests, requiring from now on at least two traits per test class: a Category and the Owning Team.

**Comment by ccfromms** [comment]
> Hi! Please, sync this PR with main. I've added new validations on top of tests, requiring from now on at least two traits per test class: a Category and the Owning Team.

Ok, I will sync this PR.

**Comment by rbans96** [comment]
@ccfromms Thanks for syncing with main :) I am wondering if this PR is still in draft or can be published as 'Ready for review'?

**Comment by ccfromms** [comment]
> @ccfromms Thanks for syncing with main :) I am wondering if this PR is still in draft or can be published as 'Ready for review'?

Yes, this PR is ready for review. I have turned it open. Thanks!

**Comment by brendankowitz** [comment]
@ccfromms would you be able to update with main again?

**Comment by ccfromms** [comment]
> @ccfromms would you be able to update with main again?

Ok, I will update it.

**Comment by LTA-Thinking** [comment]
This PR needs to be reviewed and probably updated. @EXPEkesheth I'll make a small story for next sprint to review this and get it merged.

**Review by PTaladay — APPROVED** [review]

**Pull request merged**